src/main/java/org/minions/devfund/lourdes/WhackAMole.java

```diff
@@ -0,0 +1,127 @@
+package org.minions.devfund.lourdes;
+
+/**
+ * Class that simulate the Whackamole game.
+ */
+public class WhackAMole {
+    private static final char MOLE = 'M';
+    private static final char WHACK = 'W';
+    private static final char ASTERISK = '*';
+    private int score;
+    private int molesLeft;
+    private int attemptsLeft;
+
+    private char[][] moleGrid;
+
+    /**
+     * Constructor.
+     *
+     * @param numAttempts   Specify the max number of attempt to be used in the game.
+     * @param gridDimension Define the grid dimension.
+     */
+    public WhackAMole(int numAttempts, int gridDimension) {
+        this.attemptsLeft = numAttempts;
+        this.moleGrid = new char[gridDimension][gridDimension];
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid[i].length; j++) {
+                moleGrid[i][j] = '*';
+            }
+        }
+    }
+
+    /**
+     * Method to place a Mole into the Grid.
+     *
+     * @param x Represents the X axes position in the grid.
+     * @param y Represents the Y axes position in the grid.
+     * @return True if the mole was placed succesfully
+     */
+    public boolean place(int x, int y) {
+        if (this.moleGrid[x][y] == ASTERISK) {
+            this.moleGrid[x][y] = MOLE;
+            this.molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method that perform a whack in the grid.
+     *
+     * @param x Represents the X axes position in the grid.
+     * @param y Represents the Y axes position in the grid.
+     */
+    public void whack(int x, int y) {
+        if (moleGrid[x][y] == MOLE) {
+            molesLeft--;
+            score++;
+            moleGrid[x][y] = WHACK;
+        }
+        attemptsLeft--;
+    }
+
+    /**
+     * Method that print the grid without showing the moles.
+     */
+    public void printGridToUser() {
+        for (char[] row : moleGrid) {
+            for (char cellValue : row) {
+                if (cellValue == MOLE || cellValue == ASTERISK) {
+                    System.out.print(ASTERISK);
```

Comment:
To increase the code coverage you can return String instead of void
and use a StringBuilder instead of sout.